### PR TITLE
Feature/96 refactor set additional info method for 4.0.0

### PR DESCRIPTION
**Pull request by AB006J9**

### CODE REVIEW

File: atum/src/main/scala/za/co/absa/atum/core/Accumulator.scala [hunk 1]
```diff
@@ -94,11 +94,11 @@ class Accumulator() {
   }
 
   /** Adds a key-value pair as an additional information stored in the metadata. */
-  def setAdditionalInfo(kv: (String, String), replaceIfExists: Boolean): Unit = {
-    kv match {
-      case (key, _) =>
-        if (replaceIfExists || !controlMeasure.metadata.additionalInfo.contains(kv._1)) {
-          val newInfo = controlMeasure.metadata.additionalInfo + kv
+  def setAdditionalInfo(key: String, value: String, replaceIfExists: Boolean): Unit = {
+    key match {
```
Comment (AB006J9) [inline atum/src/main/scala/za/co/absa/atum/core/Accumulator.scala:98]: Get rid of match {case (key)}

**Review by benedeki — DISMISSED** [review]
Just read the code.

File: atum/src/test/scala/za/co/absa/atum/AtumImplicitsSpec.scala [hunk 2]
```diff
@@ -50,17 +50,47 @@ class AtumImplicitsSpec extends AnyFlatSpec with SparkTestBase with Matchers {
     val df = spark.read.json(Seq(inputData).toDS)
 
     // act
-    df.setAdditionalInfo(("additionalKey1", "additionalValue1"))
+    df.setAdditionalInfo("additionalKey1", "additionalValue1")
+    df.setAdditionalInfo("additionalKey1", "anotherAdditionalValue")
```
Comment (dk1844) [inline atum/src/test/scala/za/co/absa/atum/AtumImplicitsSpec.scala:54]: maybe a quick comment like `//replaceIfExists is false by default, so this value will not be applied`

or even the not-used value can signify it:
```suggestion
    df.setAdditionalInfo("additionalKey1", "theValueWillNotBeOverwrittenByThis") // replaceIfExists is false by default
```

File: atum/src/test/scala/za/co/absa/atum/AtumImplicitsSpec.scala [hunk 3]
```diff
@@ -50,17 +50,47 @@ class AtumImplicitsSpec extends AnyFlatSpec with SparkTestBase with Matchers {
...
+
+    // assert state of additionalInfo before
+    inputControlMeasure.metadata.additionalInfo shouldBe Map.empty
+
+    import spark.implicits._
+    val df = spark.read.json(Seq(inputData).toDS)
+
+    // act
+    df.setAdditionalInfo("additionalKey1", "additionalValue1")
+    df.setAdditionalInfo("additionalKey1", "updatedAdditionalValue", true)
```
Comment (dk1844) [inline atum/src/test/scala/za/co/absa/atum/AtumImplicitsSpec.scala:84]:
```suggestion
    df.setAdditionalInfo("additionalKey1", "updatedAdditionalValue", replaceIfExists = true)
```
It's correct like you wrote it, but spelling out what the boolean is could improve the readability a bit further :)

**Review by dk1844 — APPROVED** [review]
- [x] code reviewed
- [x] pulled
- [x] built
- [x] run the tests

I have also noticed that there are some unused imports in the `AtumImplicitsSpec.scala`.

Otherwise LGTM.

**Review by dk1844 — APPROVED** [review]
LGTM.

**Review by benedeki — APPROVED** [review]
Just read the code

**Pull request merged**